corehq/motech/docs/value_source.rst

```diff
@@ -0,0 +1,256 @@
+----------------------------
+
+MOTECH configurations use "value sources" to refer to values in
+CommCare, like values of case properties or form questions.
+
+
+Data Types
+^^^^^^^^^^
+
+Integrating structured data with remote systems can involve converting
+data from one format or data type to another.
+
+For standard OpenMRS properties (person properties, name properties and
+address properties) MOTECH will set data types correctly, and
+integrators do not need to worry about them.
+
+But administrators may want a value that is a date in CommCare to a
+datetime in a remote system, or vice-versa. To convert from one to the
+other, set data types for value sources.
+
+The default is for both the CommCare data type and the external data
+type not to be set. e.g. ::
+
+    {
+      "expectedDeliveryDate": {
+        "doc_type": "CaseProperty",
+        "case_property": "edd",
+        "commcare_data_type": null,
+        "external_data_type": null
+      }
+    }
+
+To set the CommCare data type to a date and the OpenMRS data type to a
+datetime for example, use the following::
+
+    {
+      "expectedDeliveryDate": {
+        "doc_type": "CaseProperty",
+        "case_property": "edd",
+        "commcare_data_type": "cc_date",
+        "external_data_type": "omrs_datetime"
+      }
+    }
+
+For the complete list of CommCare data types, see `MOTECH constants`_.
+For the complete list of DHIS2 data types, see `DHIS2 constants`_. For
+the complete list of OpenMRS data types, see `OpenMRS constants`_.
+
+
+.. _MOTECH constants: https://github.com/dimagi/commcare-hq/blob/master/corehq/motech/const.py
+.. _DHIS2 constants: https://github.com/dimagi/commcare-hq/blob/master/corehq/motech/dhis2/const.py
+.. _OpenMRS constants: https://github.com/dimagi/commcare-hq/blob/master/corehq/motech/openmrs/const.py
+
+
+Import-Only and Export-Only Values
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+In configurations like OpenMRS Atom feed integration that involve both
+sending data to OpenMRS and importing data from OpenMRS, sometimes some
+values should only be imported, or only exported.
+
+Use the ``direction`` property to determine whether a value should only
+be exported, only imported, or (the default behaviour) both.
+
+For example, to import a patient value named "hivStatus" as a case
+property named "hiv_status" but not export it, use
+``"direction": "in"``::
+
+    {
+      "hivStatus": {
+        "doc_type": "CaseProperty",
+        "case_property": "hiv_status",
+        "direction": "in"
+      }
+    }
+
+To export a form question, for example, but not import it, use
+``"direction": "out"``::
+
+    {
+      "hivStatus": {
+        "doc_type": "FormQuestion",
+        "case_property": "hiv_status",
+        "direction": "out"
+      }
+    }
+
+Omit ``direction``, or set it to ``null``, for values that should be
+both imported and exported.
+
+
+CaseProperty
+^^^^^^^^^^^^
+
+.. autoclass:: corehq.motech.value_source.CaseProperty
```

Comment:
You may also try using `automodule` with some [combination of toggles](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html) to pull in relevant docs without declaring each class here.

Reply:
:+1: Nice.